pkg/observer_test.go

```diff
@@ -0,0 +1,237 @@
+package ocr2keepers
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+type MockTick struct {
+	mock.Mock
+}
+
+func (m *MockTick) GetUpkeeps(ctx context.Context) ([]UpkeepPayload, error) {
+	ret := m.Called(ctx)
+	return ret.Get(0).([]UpkeepPayload), ret.Error(1)
+}
+
+type MockRunner2 struct {
+	mock.Mock
+}
+
+func (m *MockRunner2) CheckUpkeeps(ctx context.Context, payloads []UpkeepPayload) ([]CheckResult, error) {
+	ret := m.Called(ctx, payloads)
+	return ret.Get(0).([]CheckResult), ret.Error(1)
+}
+
+type MockPreprocessor struct {
+	mock.Mock
+}
+
+func (m *MockPreprocessor) PreProcess(ctx context.Context, payloads []UpkeepPayload) ([]UpkeepPayload, error) {
+	ret := m.Called(ctx, payloads)
+	return ret.Get(0).([]UpkeepPayload), ret.Error(1)
+}
+
+type MockPostprocessor struct {
```

Comment:
🏄 I would have these mock types as unexported but I'm happy with whatever you prefer 👍

Reply:
i didnt see a mocks directory. but i can change it later